### PR TITLE
feat(ci): add ci

### PR DESCRIPTION
**Pull request by linepro6**

### CODE REVIEW

**Comment by icestream32** [comment]
My Comment is, no problem!!!

**Review by icestream32 — APPROVED** [review]

**Review by icestream32 — APPROVED** [review]
Approve.

**Review by icestream32 — CHANGES_REQUESTED** [review]
Approve

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      - name: Build
+        run: pnpm build
+
+      - name: Set Environment
+        shell: bash
+        run: |
+          echo "branch=$(echo $GITHUB_REF | cut -d'/' -f 3)" >> "$GITHUB_ENV"
+          echo "sha_short=$(echo $GITHUB_SHA | cut -c1-7)" >> "$GITHUB_ENV"
+      
+      - uses: actions/upload-artifact@v4
```
Comment (icestream32) [inline .github/workflows/build.yml:37]: Test

**Review by icestream32 — APPROVED** [review]

**Review by icestream32 — APPROVED** [review]
Test

**Review by linepro6 — COMMENTED** [review]
checked

**Pull request merged**